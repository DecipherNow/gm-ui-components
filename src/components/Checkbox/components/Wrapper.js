import styled from "styled-components";
import { COLOR_CONTENT, FONT_STACK_BASE, FONT_SIZE_BASE } from "style/styleVariables";
import { spacingScale } from "style/styleFunctions";

const fontFamily = props => props.theme.FONT_STACK_BASE || FONT_STACK_BASE;
const fontSize = props => props.theme.FONT_SIZE_BASE || FONT_SIZE_BASE;
const color = props => props.theme.COLOR_CONTENT || COLOR_CONTENT;

const Wrapper = styled.label`
  ${props => getPosition(props.labelPosition)};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  color: ${color};
  display: flex;
  width: fit-content;
  align-items: center;
`;

const CONTROL_SPACING = spacingScale(1);

function getPosition(position) {
  switch (position) {
    case "top":
      return `
        flex-direction: column-reverse;

        > input {
          margin-top: ${CONTROL_SPACING};
        }
      `;
    case "bottom":
      return `
        flex-direction: column;

        > input {
          margin-bottom: ${CONTROL_SPACING};
        }
      `;
    case "left":
      return `
        flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
    case "right":
      return `
        flex-direction: row;

        > input {
          margin-right: ${CONTROL_SPACING};
        }
      `;
    default:
      return `
       flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
  }
}

export default Wrapper;
