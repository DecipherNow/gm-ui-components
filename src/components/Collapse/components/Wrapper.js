import styled, { css } from "styled-components";

import { keen } from "style/theme";

export const Wrapper = styled.section`
  width: 100%;
  transition: all 0.25s ease;
  min-height: 3.4rem;
  font-family: ${({ theme }) => theme.FONT_STACK_DEFAULT};
  border-top: 1px solid ${({ theme }) => theme.COLOR_KEYLINE_DEFAULT};

  ${({ isOpen }) =>
    isOpen &&
    css`
      overflow: visible;
    `};
`;

Wrapper.defaultProps = {
  theme: keen
};
