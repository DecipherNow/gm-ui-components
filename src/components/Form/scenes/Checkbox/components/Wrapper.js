import styled from "styled-components";
import { spacingScale } from "style/styleFunctions";
import { keen } from "style/styleVariables";

const Wrapper = styled.label`
  ${props => getPosition(props.labelPosition)};
  font-family: ${props => props.theme.FONT_STACK_BASE};
  font-size: ${props => props.theme.FONT_SIZE_BASE};
  color: ${props => props.theme.COLOR_CONTENT};
  display: flex;
  width: fit-content;
  align-items: center;
`;

Wrapper.defaultProps = {
  theme: keen
};

const CONTROL_SPACING = spacingScale(1);

function getPosition(position) {
  switch (position) {
    case "top":
      return `
        flex-direction: column-reverse;

        > input {
          margin-top: ${CONTROL_SPACING};
        }
      `;
    case "bottom":
      return `
        flex-direction: column;

        > input {
          margin-bottom: ${CONTROL_SPACING};
        }
      `;
    case "left":
      return `
        flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
    case "right":
      return `
        flex-direction: row;

        > input {
          margin-right: ${CONTROL_SPACING};
        }
      `;
    default:
      return `
       flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
  }
}

export default Wrapper;
