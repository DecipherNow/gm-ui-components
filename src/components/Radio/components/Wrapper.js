import styled from "styled-components";
import { FONT_STACK_DEFAULT, FONT_SIZE_TEXT_DEFAULT } from "style/theme";
import { spacingScale } from "style/styleFunctions";

const Wrapper = styled.label`
  ${props => getPosition(props.labelPosition)};
  font-family: ${FONT_STACK_DEFAULT};
  font-size: ${FONT_SIZE_TEXT_DEFAULT};
  display: flex;
  width: fit-content;
  align-items: center;
`;

const CONTROL_SPACING = spacingScale(1);

function getPosition(position) {
  switch (position) {
    case "top":
      return `
        flex-direction: column-reverse;

        > input {
          margin-top: ${CONTROL_SPACING};
        }
      `;
    case "bottom":
      return `
        flex-direction: column;

        > input {
          margin-bottom: ${CONTROL_SPACING};
        }
      `;
    case "left":
      return `
        flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
    case "right":
      return `
        flex-direction: row;

        > input {
          margin-right: ${CONTROL_SPACING};
        }
      `;
    default:
      return `
       flex-direction: row-reverse;

        > input {
          margin-left: ${CONTROL_SPACING};
        }
      `;
  }
}

export default Wrapper;
