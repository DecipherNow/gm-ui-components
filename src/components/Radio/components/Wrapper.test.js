import React from "react";
import { shallow } from "enzyme";

import Wrapper from "./Wrapper";

describe("Wrapper", () => {
  it("should render", () => {
    const aWrapper = shallow(<Wrapper />);
    expect(aWrapper).toMatchSnapshot();
  });
});
