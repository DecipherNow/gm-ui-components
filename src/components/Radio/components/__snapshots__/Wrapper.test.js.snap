// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render 1`] = `
.c0 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input {
  margin-left: 8px;
}

<label
  className="c0"
/>
`;

exports[`Wrapper should render different label positions 1`] = `
.c0 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input {
  margin-left: 8px;
}

<label
  className="c0"
/>
`;

exports[`Wrapper should render different label positions 2`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input {
  margin-right: 8px;
}

<label
  className="c0"
/>
`;

exports[`Wrapper should render different label positions 3`] = `
.c0 {
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input {
  margin-top: 8px;
}

<label
  className="c0"
/>
`;

exports[`Wrapper should render different label positions 4`] = `
.c0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input {
  margin-bottom: 8px;
}

<label
  className="c0"
/>
`;
